c/include/ml-lxm-service-internal.h

```diff
@@ -0,0 +1,273 @@
+ *     std::cout << "Failed to create prompt" << std::endl;
+ *     ml_lxm_session_destroy(session);
+ *     return -1;
+ *   }
+ *
+ *   // Add text to prompt
+ *   ret = ml_lxm_prompt_append_text(prompt, "Explain quantum computing in simple terms");
+ *   if (ret != ML_ERROR_NONE) {
+ *     std::cout << "Failed to append text to prompt" << std::endl;
+ *     ml_lxm_prompt_destroy(prompt);
+ *     ml_lxm_session_destroy(session);
+ *     return -1;
+ *   }
+ *
+ *   // 3. Generate response with custom options
+ *   ml_lxm_generation_options_s options = {
+ *     .temperature = 1.2,
+ *     .max_tokens = 128
+ *   };
+ *
+ *   std::cout << "AI Response: ";
+ *   ret = ml_lxm_session_respond(session, prompt, &options, token_handler, NULL);
+ *   if (ret != ML_ERROR_NONE) {
+ *     std::cout << "Failed to generate response" << std::endl;
+ *   }
+ *   std::cout << std::endl;
+ *
+ *   // 4. Cleanup
+ *   ml_lxm_prompt_destroy(prompt);
+ *   ml_lxm_session_destroy(session);
+ *
+ *   return 0;
+ * }
+ *
+ * static void token_handler(ml_service_event_e event,
+ *                          ml_information_h event_data,
+ *                          void *user_data) {
+ *   ml_tensors_data_h data = NULL;
+ *   void *_raw = NULL;
+ *   size_t _size = 0;
+ *   int ret;
+ *
+ *   switch (event) {
+ *   case ML_SERVICE_EVENT_NEW_DATA:
+ *     if (event_data != NULL) {
+ *       ret = ml_information_get(event_data, "data", &data);
+ *       if (ret == ML_ERROR_NONE) {
+ *         ret = ml_tensors_data_get_tensor_data(data, 0U, &_raw, &_size);
+ *         if (ret == ML_ERROR_NONE && _raw != NULL && _size > 0) {
+ *           std::cout.write(static_cast<const char *>(_raw), _size);
+ *           std::cout.flush();
+ *         }
+ *       }
+ *     }
+ *     break;
+ *   default:
+ *     break;
+ *   }
+ * }
+ * @endcode
+ */
+
+#ifndef __ML_LXM_SERVICE_INTERNAL_H__
+#define __ML_LXM_SERVICE_INTERNAL_H__
+
+#include <stdlib.h>
+#include <ml-api-service.h>
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @brief Enumeration for LXM service availability status.
+ */
+typedef enum
+{
+  ML_LXM_AVAILABILITY_AVAILABLE = 0,
+  ML_LXM_AVAILABILITY_DEVICE_NOT_ELIGIBLE,
+  ML_LXM_AVAILABILITY_SERVICE_DISABLED,
+  ML_LXM_AVAILABILITY_MODEL_NOT_READY,
+  ML_LXM_AVAILABILITY_UNKNOWN
+} ml_lxm_availability_e;
+
+/**
+ * @brief Checks LXM service availability.
+ * @param[out] status Current availability status.
+ * @return ML_ERROR_NONE on success, error code otherwise.
+ */
+int ml_lxm_check_availability (ml_lxm_availability_e * status);
+
+/**
+ * @brief A handle for lxm session.
+ */
+typedef void *ml_lxm_session_h;
+
+/**
+ * @brief Creates an LXM session.
+ * @param[out] session Session handle.
+ * @param[in] config_path Path to configuration file.
+ * @param[in] instructions Initial instructions (optional).
+ * @return ML_ERROR_NONE on success.
+ */
+int ml_lxm_session_create (ml_lxm_session_h * session, const char *config_path, const char *instructions);
```

Comment:
Param order:
Tizen native api suggests destination param (session) is last order.

Reply:
Thank you. I've fixed it.